tests/validator/validator.cpp

```diff
@@ -647,16 +691,14 @@ TEST(Validator, invalidMathMLCiAndCnElementsWithCellMLUnits)
         "  </apply>\n"
         "</math>\n";
     const std::vector<std::string> expectedErrors = {
-        "Math in component 'componentName' contains 'B' as a bvar ci element but it is already a variable name.",
-        "Math has a cn element with a cellml:units attribute 'invalid' that is not a valid reference to units in component 'componentName' or a standard unit.",
-        "Math ci element has an invalid attribute type 'value' in the cellml namespace.",
-        "MathML ci element has an empty child element.",
-        "MathML ci element has the child text 'undefined_variable' which does not correspond with any variable names present in component 'componentName' and is not a variable defined within a bvar element.",
-        "MathML ci element has no child.",
+        "Math cn element has an invalid attribute type 'value' in the cellml namespace.  Attribute 'units' is the only CellML namespace attribute allowed.",
+        "Math has a cn element with a cellml:units attribute 'invalid' that is not a valid reference to units in the model 'modelName' or a standard unit.",
+        "MathML ci element has the child text 'new_bvar' which does not correspond with any variable names present in component 'componentName'.",
+        "MathML ci element has the child text 'undefined_variable' which does not correspond with any variable names present in component 'componentName'.",
         "CellML identifiers must contain one or more basic Latin alphabetic characters.",
         "Math cn element with the value '2.0' does not have a valid cellml:units attribute.",
-        "W3C MathML DTD error: Namespace prefix cellml for value on ci is not defined.",
-        "No declaration for attribute cellml:value of element ci.",
+        "W3C MathML DTD error: No declaration for attribute units of element ci.",
```

Comment:
Error message is ambiguous ... not sure whether it's saying we must or must not have units on ci

Reply:
It's a message from libXml2 when using the MathML DTD we don't have any control over what is says.